src/mds/Server.cc

```diff
@@ -5722,6 +5722,47 @@ int Server::parse_layout_vxattr(string name, string value, const OSDMap& osdmap,
   return 0;
 }
 
+int64_t Server::parse_human_readable_size(string_view value_str)
+{
+  if (value_str.empty()) {
+    return -CEPHFS_EINVAL;
+  }
+  if (value_str.front() == '-') {
+    dout(10) << "negative value not permitted" << dendl;
+    return -CEPHFS_EINVAL;
+  }
+  unordered_map<char, int64_t> size_values = {{'k', 1L << 10}, {'m', 1L << 20}, 
+                                             {'g', 1L << 30}, {'t', 1L << 40}};
+  char data_size = tolower(value_str.back());
```

Comment:
nit: use `auto`

Reply:
maybe add an additional check if the suffix is a valid char. At this point you could also validate it with `size_values`.

Reply:
> maybe add an additional check if the suffix is a valid char. At this point you could also validate it with size_values.

This is going to be checked and caught in the below if-else block. It suffix is not in map we can assert that it's not a valid char

Reply:
> nit: use auto

done.